scripts/js/lib/api/generateApiComponents.ts

```diff
@@ -252,7 +244,8 @@ function prepareAttributeOrPropertyProps(
     modifiers: filteredModifiers,
   };
 
-  if (!priorApiType && id) {
+  const pageHeading = $dl.siblings("h1").text();
+  if (id.endsWith(pageHeading) && pageHeading != "") {
```

Comment:
When is pageHeading ever `""`? Ah, probably unit tests?

Reply:
It could be an empty string if for some reason (structure of the page) there isn't any `<h1>` as a sibling. I'm not sure we are testing the case explicitly, but it's implicitly tested in some cases like in this test where it doesn't have a heading.

https://github.com/Qiskit/documentation/blob/3c0b8ebd14b214025ee2480ca36cf750f74cffed/scripts/js/lib/api/processHtml.test.ts#L570

If we had an `<h1>` with the same name as the attribute in the test, the result wouldn't include the `<h3>` because it would enter into the if of line 248.

Reply:
I applied the suggestion to the Function component too. We were using the same conditional. Thanks :)